### DISCUSSION TITLE
Slingbox Solo error python

### DISCUSSION THREAD

**Discussion by elgato45** [GerryDazoo/Slinger#542] [Q&A]
Hi guys, I have a trouble run the server in windows and a raspberry pi, this is the error when a player connect to a server.
thanks for your help.

`Broadcast Responder Running
02/24/2023, 18:42:01.752   Streaming request from ('192.168.1.134', 54227)  /slingbox 
STREAM=192.168.1.134:54227:0
SlingStudio Using encrypted password: XXXXXXXXX

Slinginfo  "Solo/Pro/ProHD" 5 30 192.168.1.125 5001 3000 10 password
SlingStudio Starting Stream for  192.168.1.134:54227
SlingStudio Opening Control stream 0x2000 0 0
Connecting... ('192.168.1.125', 5001) Control
cmd: 0x67 err: 0x2b 8
Traceback (most recent call last):
  **File "slingbox_server.py", line 836, in streamer
    s_ctl, stream  = start_slingbox_session(streams)
  File "slingbox_server.py", line 367, in start_slingbox_session
    elif start == 0x2b:
NameError: name 'start' is not defined**
SlingStudio Badness starting slingbox session  name 'start' is not defined None
Streamer:  SlingStudio Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox`

**Comment by GerryDazoo** [comment]
This "name" error is a bug BUT it's a side effect of your slingbox not responding to the very first message. I suspect the box is in a bad state. Please power cycle the slingbox. But your Solo might be bricked if you haven't blocked outgoing TCP port 5678 for the Slingbox. 
Next time please post ALL the server output.

**Reply by MicroXs** [reply]
I have blocked 5678 and still have the error. Is it related to the model 240?

**Reply by MicroXs** [reply]
Version : 4.0 Running on Linux-4.19.66-v7+-armv7l-with-debian-9.13 pid= 23696 ./slingbox_server.py
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Building page for Slingbox
Using built in default remote page definition.
Starting Streamer Thread for Slingbox
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
SlingMicroXs Checking for slingbox at ('192.168.0.110', 5001) 0
SlingMicroXs ('192.168.0.110', 5001) OK
SlingMicroXs Using slingbox at  ('192.168.0.110', 5001)
Streamer:  SlingMicroXs Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app "slingbox_server" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
Broadcast Responder Running
Server IP address is 192.168.0.134
02/26/2023, 15:34:12.86   Streaming request from ('192.168.0.153', 57137)  /slingbox
STREAM=192.168.0.153:57137:0

Slinginfo  "240" 12 30 192.168.0.110 5001 3072 10 xxxxxxx
SlingMicroXs Starting Stream for  192.168.0.153:57137
SlingMicroXs Opening Control stream 0x2000 0 0
Connecting... ('192.168.0.110', 5001) Control
cmd: 0x67 err: 0x2b 8
Traceback (most recent call last):
  File "./slingbox_server.py", line 836, in streamer
    s_ctl, stream  = start_slingbox_session(streams)
  File "./slingbox_server.py", line 367, in start_slingbox_session
    elif start == 0x2b:
NameError: name 'start' is not defined
SlingMicroXs Badness starting slingbox session  name 'start' is not defined None
Streamer:  SlingMicroXs Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
When you say you blocked TCP port 5678 when did you do that? Before connecting your Solo to your network? Looking at the code it sure looks like you Solo is bricked.

**Reply by MicroXs** [reply]
I blocked TCP/UDP 5678 after the errors. Rebooted the Slingbox. Tried again. It means the Slingbox is bricked forever?

**Reply by elgato45** [reply]
Hi Gerry, thanks for all your effort, I have two slingbox solo and have the same error keeps appearing, I  power cycle the slingbox it several times but the error does not change. But the two boxes were working perfectly since before they shutdown the service. 

Slinginfo  "Solo/Pro/ProHD" 5 30 192.168.1.120 5002 3000 10 <password>
**SlingRoom** Starting Stream for  192.168.1.134:64866
**SlingRoom** Opening Control stream 0x2000 0 0
Connecting... ('192.168.1.120', 5002) Control
cmd: 0x67 err: 0x2b 8
Traceback (most recent call last):
  File "slingbox_server.py", line 836, in streamer
    s_ctl, stream  = start_slingbox_session(streams)
  File "slingbox_server.py", line 367, in start_slingbox_session
    elif start == 0x2b:
NameError: name 'start' is not defined
SlingRoom Badness starting slingbox session  name 'start' is not defined None
Streamer:  SlingRoom Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox

**Reply by GerryDazoo** [reply]
I think you Solo is bricked.
Try this version, it fixes the exception but it won't fix your box.
[slingbox_server.zip](https://github.com/GerryDazoo/Slinger/files/10863880/slingbox_server.zip)

**Reply by sp222222** [reply]
it's a shame five months later and folks still bricking by not reading up beforehand... I bricked 3 myself N9.

**Reply by RocketAlf** [reply]
...and yet it happened and I guess it still happens :-o. I have two Solo broken on N9. And no genius has yet found out how to revive the Solo :-(.
Fortunately, the revival succeeds with the ProHD :-). I have now bought a switch GS308T to be able to lock port 5678 (my Internet router does not support this). Interesting is the log ...every 5 minutes the ProHD tries to phone home via port 5678. It does not give up :-/.
